### PR TITLE
bpo-36144: Document PEP 584.

### PR DESCRIPTION
**Pull request by brandtbucher**
https://bugs.python.org/issue36144

### CODE REVIEW

File: Doc/library/stdtypes.rst [hunk 1]
```diff
@@ -4392,6 +4392,22 @@ pairs within braces, for example: ``{'jack': 4098, 'sjoerd': 4127}`` or ``{4098:
          >>> d.values() == d.values()
          False
 
+   .. method:: d | other
```
Comment (gvanrossum) [inline Doc/library/stdtypes.rst]: Maybe use ‘describe’ instead of‘method’. (What do the docs for ‘set’ use?)

File: Doc/whatsnew/3.9.rst [hunk 2]
```diff
@@ -70,6 +70,51 @@ Summary -- Release highlights
 New Features
 ============
 
+Dictionary Merge & Update Operators
+-----------------------------------
+
+Merge (``|``) and update (``|=``) operators have been added to the built-in
+:class:`dict` class.
+
+.. XXX The rest is just a copy-paste of the PEP's Specification section.
```
Comment (gvanrossum) [inline Doc/whatsnew/3.9.rst]: Maybe drop the rest and link to the PEP instead? The comment above praises brevity. :-)

File: Doc/library/stdtypes.rst (hunk 1 above)
Comment (brandtbucher) [reply Doc/library/stdtypes.rst]: `set` uses `method`, but it lists both the method and operator together. I agree that `describe` is probably better here.

File: Doc/whatsnew/3.9.rst (hunk 2 above)
Comment (brandtbucher) [reply Doc/whatsnew/3.9.rst]: I think the brevity bit is about the summary section, no? I latched onto "Do not spend very much time on the wording of your changes"... hence, the copy-paste. But you're right that it probably does not add much here.

File: Doc/whatsnew/3.9.rst [hunk 3]
```diff
@@ -70,6 +70,17 @@ Summary -- Release highlights
 New Features
 ============
 
+Dictionary Merge & Update Operators
+-----------------------------------
+
+Merge (``|``) and update (``|=``) operators have been added to the built-in
+:class:`dict` class.
+
+.. XXX https://www.python.org/dev/peps/pep-0584/
```
Comment (gvanrossum) [inline Doc/whatsnew/3.9.rst]: I'm not sure why you would want to keep a full link to the PEP in a comment. Do other sections do this?

File: Doc/whatsnew/3.9.rst [hunk 4]
```diff
@@ -70,6 +70,17 @@ Summary -- Release highlights
...
+
+See :pep:`584` for a full description.
+
+(Contributed by Brandt Bucher in :issue:`36144`.)
```
Comment (gvanrossum) [inline Doc/whatsnew/3.9.rst]: Hm, looking at the rendered docs, and comparing other sections, I wonder if this shouldn't just be a single paragraph.

File: Doc/whatsnew/3.9.rst (hunk 3 above)
Comment (brandtbucher) [reply Doc/whatsnew/3.9.rst]: I misunderstood your prior review comment!

**Review by native-api — APPROVED** [review]

**Comment by brandtbucher** [comment]
@gvanrossum I'm a bit confused. Looking at the "New Features" section of the [What's New for 3.8](https://raw.githubusercontent.com/python/cpython/master/Doc/whatsnew/3.8.rst), I see somewhat detailed docs with examples spread across several paragraphs (for example, for assignment expressions and positional-only parameters). The smaller, non-PEP items below are more compact though.

However, I'm probably overthinking this, which is exactly what the comments at the top warn against. 🙂

I'll just keep the short version and the notes in one paragraph, as you suggest, and  Łukasz or whoever will go back later and expand as needed, then?

**Review by gvanrossum — APPROVED** [review]
It's true, we could add more. But the feature is so simple, I think it doesn't need more. (It's not new syntax.)

**Comment by gvanrossum** [comment]
I'll land it now. If you feel regrets later (or someone else thinks the whatsnew section on dict|dict should have more information), just submit a PR with a doc update, referencing the same issue (https://bugs.python.org/issue36144).

**Pull request merged**